api/query_engine.go

```diff
@@ -111,6 +125,30 @@ func alignRequests(reqs []models.Req, aggSettings mdata.AggSettings) ([]models.R
 
 	chosenInterval := options[selected].interval
 
+	// by default we select the lowest res (longest time range) option.
+	selectedHighestRes := len(options) - 1
+	// then we loop over the remaining options trying to find the highest res where the
+	// oldest timestamp is still older than our From ts
+	for i := len(options) - 2; i >= 0; i-- {
+		if oldestTs(options[i].ttl) > reqs[0].From {
+			break
+		}
+		selectedHighestRes = i
+	}
+
+	// record how many datapoints will be sent to graphite-api if we use the old-style selection mechanism
+	dpAdjGranularity.SetUint32(tsRange / chosenInterval)
+
+	// record the aggregation index that has been selected
+	aggAdjGranularity.SetUint32(uint32(selected))
```

Comment:
also one confusing thing is that the index reported is in the filtered list of archives (after removing non-ready archives).
I think it'll be more intuitive and consistent over time if we just report the index based on what `agg-settings` says.